
# ## TOPIC MODIFIES A FRAME COMPONENT
# unity:
#   type: "frame"
#   target_frame: "base_link"
#   component: "transform"
#   modifier: "set_odom"

# ## TOPIC INSTANTIATE CHILD GAMEOBJECTS
# unity: #spawn series of blocks relative to laserscanner
#   type: "child"
#   target_frame: "laser_scanner"
#   component: ""
#   modifier: "show_laser_scan"

# ## TOPIC SHARES DATA
# unity: #pushes image onto assigned target ui
#   type: "data"
#   target_frame: ""
#   component: "" #we could make it point to a material mqtt_reference.mat?
#   modifier: "set_image"








topic_list:

## TOPIC MODIFIES A FRAME COMPONENT
- topic: "/odom"
  msg_type: "nav_msgs/Odometry"
  unity:
    type: "frame"
    target_frame: "odom"
    component: "UnityEngine.Transform, UnityEngine.CoreModule"
    modifier: "SetOdom"
  mqtt:
    frequency: 20


# ## TOPIC MODIFIES A NON-FRAME COMPONENT
# - topic: "/map"
#   msg_type: "nav_msgs/OccupancyGrid"
#   unity:
#     target_frame: "RobotModel"
#     component: "UnityEngine.Transform, UnityEngine.CoreModule"
#     modifier: "SetOdom"
#   mqtt:
#     frequency: 20


# - topic: "/camera/rgb/image_raw"
#   msg_type: "sensor_msgs/Image"
#   unity:
#     type: "frame"
#     target_frame: "Display_3_panel"
#     component: "UnityEngine.Renderer, UnityEngine.CoreModule"
#     modifier: "SetImageMaterial"
#   mqtt:
#     frequency: 1


# ## TOPIC MODIFIES A FRAME CHILD OBJECT
# - topic: "/robot_model"
#   msg_type: "visualization_msgs/Marker[]"
#   unity:
#     target_frame: "RobotModel"
#     modifier: "SetModel"
# - topic: "/move_base/current_goal"
#   msg_type: "geometry_msgs/PoseStamped"
#   unity:
#     target_frame: "GoalMarker"
#     modifier: "SetPoseStamped"
#   mqtt:
#     latched: true  
# - topic: "/move_base/NavfnROS/plan"
#   unity:
#     target_frame: ""
#     component: ""
#     modifier: ""
#   mqtt:
#     latched: true
# - topic: "/scan"
#   msg_type: "sensor_msgs/LaserScan"
#   unity:
#     target_frame: "LaserScanner"
#     modifier: "LaserScan"
